src/components/grouped-character/grouped-character.pw.tsx

```diff
@@ -0,0 +1,496 @@
+
+      const inputParent = getDataElementByValue(page, "input")
+        .locator("..")
+        .locator("..");
+      await expect(inputParent).toHaveCSS("max-width", maxWidth);
+    });
+  });
+
+  test(`should render with max-width of 100%, when maxWidth prop is passed with no value`, async ({
+    mount,
+    page,
+  }) => {
+    await mount(<GroupedCharacterComponent maxWidth="" />);
+
+    const inputParent = getDataElementByValue(page, "input")
+      .locator("..")
+      .locator("..");
+    await expect(inputParent).toHaveCSS("max-width", "100%");
+  });
+
+  ([
+    [[1, 1, 4], "123", "123", "1-2-3"],
+    [[3, 2, 1], "sage123", "sage12", "sag-e1-2"],
+    [[3, 3, 3], "123testtest", "123testte", "123-tes-tte"],
+    [[4, 1, 2], "1234567", "1234567", "1234-5-67"],
+    [[5, 2, 2], "9876543211", "987654321", "98765-43-21"],
+    [[1, 1, 3], "123456789", "12345", "1-2-345"],
+  ] as [GroupedCharacterProps["groups"], string, string, string][]).forEach(
+    ([groups, inputValue, rawValue, formattedValue]) => {
+      test(`should call onChange callback when a type event is triggered using ${groups} as groups and ${inputValue} as inputValue and return ${rawValue} as formattedValue`, async ({
+        mount,
+        page,
+      }) => {
+        let callbackCount = 0;
+        await mount(
+          <GroupedCharacterComponent
+            onChange={() => {
+              callbackCount += 1;
+            }}
+            groups={groups}
+          />
+        );
+
+        const input = getDataElementByValue(page, "input");
+        await input.fill(inputValue);
+        await input.blur();
+        expect(callbackCount).toBe(1);
+        await expect(input).toHaveAttribute("value", formattedValue);
+      });
+    }
+  );
+
+  test(`should call onBlur callback when a blur event is triggered`, async ({
+    mount,
+    page,
+  }) => {
+    let callbackCount = 0;
+    await mount(
+      <GroupedCharacterComponent
+        onBlur={() => {
+          callbackCount += 1;
+        }}
+      />
+    );
+
+    const input = getDataElementByValue(page, "input");
+    await input.fill("1");
+    await input.blur();
+    expect(callbackCount).toBe(1);
+  });
+
+  test(`should have the expected border radius styling`, async ({
+    mount,
+    page,
+  }) => {
+    await mount(<GroupedCharacterComponent />);
+
+    await expect(getElement(page, "input")).toHaveCSS("border-radius", "4px");
+  });
+});
+
+test.describe("Accessibility tests for GroupedCharacter", () => {
+  test(`should pass accessibility tests for GroupedCharacter Default example`, async ({
+    mount,
+    page,
+  }) => {
+    await mount(<DefaultStory />);
+
+    await checkAccessibility(page);
+  });
+
+  ([
+    SIZE.SMALL,
+    SIZE.MEDIUM,
+    SIZE.LARGE,
+  ] as GroupedCharacterProps["size"][]).forEach((size) => {
+    test(`should pass accessibility tests for GroupedCharacter Sizes ${size} size example`, async ({
```

Comment:
non-blocking: maybe we can get rid of one word size and change it to `should pass accessibility tests for GroupedCharacter ${size} size example`, it looks a bit duplicated when we have two "size" word